.github/workflows/docker-new.yml

```diff
@@ -0,0 +1,134 @@
+            echo "TAG=latest" >> $GITHUB_ENV
+          elif echo "$GITHUB_REF" | grep -q '^refs/tags/'; then
+            # Pushes tag - deploy tag name.
+            echo "TAG=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
+          else
+            # Use commit SHA for PR
+            echo "TAG=${GITHUB_SHA::8}" >> $GITHUB_ENV
+          fi
+      - name: Build and push to local registry
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: Dockerfile-CI.Dockerfile
+          platforms: ${{ env.DOCKER_PLATFORMS }}
+          push: true
+          tags: localhost:5000/${{ env.DOCKER_REPO }}:${{ env.TAG }}
+      - name: Test docker images locally
+        run: |
+          for platform in ${DOCKER_PLATFORMS/,/ }; do
+              image="localhost:5000/${DOCKER_REPO}:${TAG}"
+              msg="Testing docker image $image on platform $platform"
+              line="${msg//?/=}"
+              printf "\n${line}\n${msg}\n${line}\n"
+              docker pull -q --platform "$platform" "$image"
+              echo -n "Image architecture: "
+              docker run --rm --entrypoint /bin/sh "$image" -c 'uname -m'
+              version=$(docker run --rm "$image" --version)
+              echo "lychee version: $version"
+              if [[ $TAG != "latest" ]] &&
+              [[ $TAG != "$version" ]] &&
+              ! echo "$version" | grep -q "$TAG"; then
+                echo "Version mismatch: lychee $version tagged as $TAG"
+                exit 1
+              fi
+          done
+      - name: Check GitHub settings
+        if: >
+          github.event_name == 'push' &&
+          (github.ref == 'refs/heads/master' ||
+          startsWith(github.ref, 'refs/tags/')) &&
+          github.repository == 'lycheeverse/lychee'
+        run: |
+          missing=()
+          [[ -n "${{ secrets.DOCKER_USERNAME }}" ]] || missing+=(DOCKER_USERNAME)
+          [[ -n "${{ secrets.DOCKER_PASSWORD }}" ]] || missing+=(DOCKER_PASSWORD)
+          for i in "${missing[@]}"; do
+            echo "Missing github secret: $i"
+          done
+          (( ${#missing[@]} == 0 )) || exit 1
+          echo "DOCKER_DEPLOY_IMAGES=true" >> $GITHUB_ENV
+      - name: Login to DockerHub
+        if: ${{ env.DOCKER_DEPLOY_IMAGES == 'true' }}
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Push images to DockerHub
+        if: ${{ env.DOCKER_DEPLOY_IMAGES == 'true' }}
+        run: |
+          image_src="${DOCKER_REPO}:${TAG}"
+          image_dst="${{ secrets.DOCKER_USERNAME }}/${{ env.DOCKER_REPO }}:${TAG}"
+          msg="Copy multi-arch docker images to DockerHub ($image)"
+          line="${msg//?/=}"
+          printf "\n${line}\n${msg}\n${line}\n"
+          skopeo copy --all --src-tls-verify=0 docker://localhost:5000/$image_src docker://docker.io/$image_dst
```

Comment:
Never used skopeo before. What's the benefit over `docker push`? Is it because the Docker daemon isn't running and we can still push to the registry?